packages/react-hooks/use-local-storage/useLocalStorage.ts

```diff
@@ -0,0 +1,75 @@
+import { Dispatch, SetStateAction, useCallback, useEffect, useState } from "react";
+
+const storeValue = async <T>(key: string, value: T): Promise<void> => {
+  if (typeof window === "undefined") {
+    return;
+  }
+  try {
+    const valueSerialized = JSON.stringify(value);
+    window.localStorage.setItem(key, valueSerialized);
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+/**
+ * Persist the state with local storage so that it remains after a page refresh. This can be useful
+ * for a dark theme or to record session information. This hook is used in the same way as useState
+ * except that you must pass the storage key in the 1st parameter. If the window object is not
+ * present (as in SSR), useLocalStorage() will return the default value.
+ *
+ * @param key - Key to use in local storage to store value
+ */
+const useLocalStorage = <T>(key: string, initialValue: T): [T, Dispatch<SetStateAction<T>>] => {
+
+  const getStoredValue = useCallback(<T>(key: string, initialValue: T): T => {
```

Comment:
since we have the function inside the hook, we can remove the params from here and the use callback

```suggestion
  const getStoredValue = () => {
```

Reply:
Should the useCallback remain so as not to recreate the getStoredValue each time it is re-rendered?